### PR TITLE
T-20805 - Get nft.trades to work with Trino SQL

### PR DESCRIPTION
**Pull request by belen-pruvost**
Our attempt with @jkylling to make `nft.trades` work on Trino SQL.

The problem laid mainly on implicit casting of types that Databricks was doing. We tested that the new super explicit view definition runs nicely on Trino.

### CODE REVIEW

**Review by jeff-dude — APPROVED** [review]
sounds good @belen-pruvost 🤝

**Comment by soispoke** [comment]
I'm worried this table UNIONed with other table in nft.trades will fail if we include the changes introduced by this PR, since the other ones will still have columns of type `string` ?

**Comment by belen-pruvost** [comment]
> I'm worried this table UNIONed with other table in nft.trades will fail if we include the changes introduced by this PR, since the other ones will still have columns of type `string` ?

@soispoke that's a good point, maybe I can include similar changes on this PR for the other tables (`magiceden`, `looksrare_ethereum `, etc). WDYT?

**Comment by soispoke** [comment]
@belen-pruvost Yes, I think it's more work but worth doing, making sure we have consistent types across nft.trades models now becomes super crucial since we can't rely on Databricks SQL to implicitly standardize it (which is actually probably a good thing to take care of anyways).

**Review by dot2dotseurat — APPROVED** [review]

**Comment by denz-e** [comment]
I'm not 100% sure if it's because of this change, but some values seem incorrect now, e.g. OpenSea's platform_fee_percentage is now returning 3 instead of 2.5

File: models/opensea/ethereum/opensea_ethereum_events.sql [hunk 1]
```diff
@@ -36,11 +36,11 @@ FROM
                 platform_fee_amount_raw,
                 platform_fee_amount,
                 platform_fee_amount_usd,
-                platform_fee_percentage,
+                CAST(platform_fee_percentage AS DECIMAL(38,0)) platform_fee_percentage,
```
Comment (jkylling) [inline models/opensea/ethereum/opensea_ethereum_events.sql:39]: Probably this and other casts of percentages should be to Double and not Decimal(38, 0).

File: models/opensea/ethereum/opensea_ethereum_events.sql [hunk 2]
```diff
@@ -77,11 +77,11 @@ FROM
                 ,platform_fee_amount_raw
                 ,platform_fee_amount
                 ,platform_fee_amount_usd
-                ,case when amount_raw > 0 then platform_fee_amount_raw / amount_raw * 100 end platform_fee_percentage
+                ,case when amount_raw > 0 then CAST ((platform_fee_amount_raw / amount_raw * 100) AS DECIMAL(38,0)) end platform_fee_percentage
```
Comment (jkylling) [inline models/opensea/ethereum/opensea_ethereum_events.sql:80]: Should be Double here

File: models/opensea/ethereum/opensea_ethereum_events.sql (hunk 1 above)
Comment (belen-pruvost) [reply models/opensea/ethereum/opensea_ethereum_events.sql:39]: I'll fix this 👍

**Comment by belen-pruvost** [comment]
> I'm not 100% sure if it's because of this change, but some values seem incorrect now, e.g. OpenSea's platform_fee_percentage is now returning 3 instead of 2.5

@denz-e thanks for reporting this - I have a fix incoming!

**Comment by denz-e** [comment]
> > I'm not 100% sure if it's because of this change, but some values seem incorrect now, e.g. OpenSea's platform_fee_percentage is now returning 3 instead of 2.5
> 
> @denz-e thanks for reporting this - I have a fix incoming!

awesome!

**Pull request merged**